NetCoreRestApi/DataLayer.EF/Repositories/BaseRepository.cs

```diff
@@ -8,46 +8,55 @@
 
 namespace DataLayer.EF.Repositories
 {
-    public class BaseRepository<TModel, TEntity, TId> : IRepository<TModel, TId>
+    public abstract class BaseRepository<TModel, TEntity, TId> : IRepository<TModel, TId>
         where TModel : BaseModel<TId>
         where TEntity : BaseEntity<TId>
     {
         protected IConverter<TEntity, TModel> Сonverter { get; private set; }
-        private DbSet<TEntity> _dbSet;
+        protected DbSet<TEntity> DbSet { get; private set; }
 
         public BaseRepository(IDbContext dbContext, IConverter<TEntity, TModel> converter)
         {
             Сonverter = converter;
-            _dbSet = dbContext.GetDbSet<TEntity>();
+            DbSet = dbContext.GetDbSet<TEntity>();
         }
 
         public virtual TModel GetById(TId id)
         {
-            var entity = _dbSet.FirstOrDefault(entity => entity.Id.Equals(id));
+            var entity = DbSet.FirstOrDefault(entity => entity.Id.Equals(id));
             return Сonverter.ConvertTo(entity);
         }
 
         public virtual IQueryable<TModel> GetAll()
         {
-            var entities = _dbSet.ToList();
+            var entities = DbSet.ToList();
```

Comment:
Make pagination for including page number, total pages, total items. Create classes for that

Reply:
itemsPerPage

Reply:
Create addition method

PageItemData GetAll(FilterData)